types/react/test/hooks.tsx

```diff
@@ -158,6 +160,7 @@ function useEveryHook(ref: React.Ref<{ id: number }>|undefined): () => boolean {
     }, []);
     React.useEffect(() => {
         dispatch({ type: 'getOlder' });
```

Comment:
I think this dispatch would now also allow zero arguments. Since this is the most common use case it should be fully supported instead of a nichè case. 

I think we need an overload to make zero-arguments dispatch safe.

Reply:
Any idea how to implement the overload?

Reply:
I haven't put any thought into it. As I said it's a very rare use case. Maybe dispatch `undefined` explicitly?

Reply:
The dispatch type is reliant on the reducer, so if the reducer is 1-ary, then the dispatch necessarily needs to be 0-ary, @dancerphil you can start with that.

Reply:
Thank a lot, i am working on it: https://gist.github.com/dancerphil/060d5aeca3c47be2e5d9ba368796b43f

I fully agree that we should not allow both defined and undefined values.